### ISSUE TITLE
Only 4 ports available

### ISSUE THREAD

**Issue by zerotohero** [marcan/lsirec#5]
Hey, thanks for making these tools! I'm running into an issue. I got a Fujitsu D2607 like you, and I've run into the same issue where only half of the ports are usable. What ended up being the fix? I used your SBR modification tool and wrote it to the card, then I believe I successfully flashed the 9211-8i p20 IT mode firmware onto the card (2118it.bin). 

I'm thinking it might have something to do with the SBR, because whenever I try to use your tool to read it back, I don't get text like I got the first time. It's like a garbled binary. Also whenever I try to use megarec to read it, it just hangs. 

Doesn't seem like I've bricked anything since the 4 ports still work, but something is definitely wrong. Any suggestions?

**Comment by Damme** [comment]
Hello

Check this out : https://marcan.st/2016/05/crossflashing-the-fujitsu-d2607/

**Comment by bronkoo** [comment]
Hello
thanks for your work!
I'm running into same issue, only 4 ports available, means only one of both Fujitsu D2607 connector is active (above one),  after doing exactly the steps below:

> UNTESTED procedure to convert from MegaRAID to IT/IR firmware or recover a bricked card

But, "that lone little mystery yellow byte" was already on '04' in my MegaRAID SBR backup file `sbr_backup.bin`, certainly since my card is a Fujitsu D2607.

The diff between my `sbr_new.bin` and your provided `sbrfj.bin` is your SAS address section blanked out, as described by you in:

[Crossflashing the Fujitsu D2607](https://marcan.st/2016/05/crossflashing-the-fujitsu-d2607/)

> Demystifying the SBR.


Finally I've flashed the `LSI-P20-2118it.bin` and `LSI-P20-mptsas2.rom` (optionally) successfully.

Any hints for me?

**Comment by bronkoo** [comment]
Ok, **both connectors available** now.

This is the used card description by dealer:

**Fujitsu SAS/SATA Raid Controller D2607-A21 GS1 PCI Express x8** 

Found these:
[https://www.davewinfield.com/cross-flashing-a-fujitsu-d2607-raid-controller/](https://www.davewinfield.com/cross-flashing-a-fujitsu-d2607-raid-controller/)

and redone your steps from 
> UNTESTED procedure to convert from MegaRAID to IT/IR firmware or recover a bricked card

but at first write `SBR-A21.bin` (my card) from link above where only the "yellow byte" is '00' (something new beside '04' and '07' and remember, my `sbr_backup.bin` has had '04') and checksum 'af' for sure, the rest is the same as your `sbrfj.bin`

Finished the rest of your steps and it ended up with same issue again, only one connector available ...

Without any reboot following the description from link above and finally rewrote SBR with `SBR-A11.bin` which is exactly the same as yours `sbrfj.bin` (check diff to my `sbr_new.bin` post before).

I'm confused about this but 8 drives on 2 connectors available now ...
Will double check later on second card.

**Comment by marcan** [comment]
Yeah, it's starting to sound like that byte has nothing to do with it and the fix for the extra connector is something else... but it seems somewhat random and I'm not sure what does it :/

**Comment by LsHallo** [comment]
A bit late, but any solution found so far? Also only having four ports available.
Tried all cable combinations but it's always only 4 drives recognized...

Tried this guide: https://forums.unraid.net/topic/57815-fujitsu-d2607-8i-crossflash-to-lsi-9211-8i/?tab=comments#comment-568494
With your sbrfj.bin and the one provided in the guide without success.
It's the A11 revision on my card.

lshw:
```
        *-pci:1
             description: PCI bridge
             product: Family 17h (Models 00h-0fh) PCIe GPP Bridge
             vendor: Advanced Micro Devices, Inc. [AMD]
             physical id: 3.1
             bus info: pci@0000:00:03.1
             version: 00
             width: 32 bits
             clock: 33MHz
             capabilities: pci pm pciexpress msi ht normal_decode bus_master cap_list
             configuration: driver=pcieport
             resources: irq:26 ioport:f000(size=4096) memory:f7500000-f7afffff
           *-raid
                description: RAID bus controller
                product: SAS2008 PCI-Express Fusion-MPT SAS-2 [Falcon]
                vendor: Broadcom / LSI
                physical id: 0
                bus info: pci@0000:06:00.0
                version: 02
                width: 64 bits
                clock: 33MHz
                capabilities: raid pm pciexpress vpd msi msix bus_master cap_list rom
                configuration: driver=mpt3sas latency=0
                resources: irq:55 ioport:f000(size=256) memory:f7a00000-f7a03fff memory:f7980000-f79bffff memory:f7500000-f757ffff memory:f79c0000-f79fffff memory:f7580000-f797ffff
```

I've since tried to edit the SBR and replace the 04 with 07 and 00 (obv. adjusted the checksum) both without success.
But due to a mistake I've lost the original SBR of my HBA. So i cannot provide it here.

**Comment by nanaya** [comment]
FWIW, I encountered this problem today and managed to "fix" it from hint in the [provided link above]( https://www.davewinfield.com/cross-flashing-a-fujitsu-d2607-raid-controller/): reflashing the SBR after flashing the firmware/bios (so the order was sbr &rarr; firmware/bios &rarr; sbr again). I used the `lsirec` from here instead of `megarec`. The image is [this blog post](https://marcan.st/2016/05/crossflashing-the-fujitsu-d2607/) (`sbrfj.bin`).

Some restarts/power cycles and head scrathings were involved so I'm not sure if there's anything else needed to be done in between.

**Comment by bronkoo** [comment]
in continuation of  https://github.com/marcan/lsirec/issues/5#issuecomment-570733873
> I'm confused about this but 8 drives on 2 connectors available now ...
> Will double check later on second card.

Time for my second HBA and again: sbr → firmware/bios → sbr  works like a charm.

**Comment by marcan** [comment]
That might explain why I thought that byte was it; I probably flashed a different SBR, saw it not work, diffed with the fujitsu one, saw the changed byte, flashed that, it worked, and wrongly concluded that was it.

Then the question is what in the SBR changes... can someone take an SBR dump *before* the second flash?

**Comment by bronkoo** [comment]
Just realized, SBR rewrite on second try here https://github.com/marcan/lsirec/issues/5#issuecomment-821111523 was `SBR-A21.bin` too...

Summarized:
Write/Flash Order:

HBA1: `SBR-A21.bin` → `LSI-P20-2118it.bin` → `LSI-P20-mptsas2.rom` → `SBR-A11.bin`
HBA2: `SBR-A21.bin` → `LSI-P20-2118it.bin` → `LSI-P20-mptsas2.rom` → `SBR-A21.bin`

Booth working fine.

Check https://github.com/marcan/lsirec/issues/5#issuecomment-570733873 for diffs between SBRs.

**Comment by ale-trevizoli** [comment]
This procedure really works. I was with same problem and did it and problem resolved.

Summarized:
Write/Flash Order:

HBA1: SBR-A21.bin → LSI-P20-2118it.bin → LSI-P20-mptsas2.rom → SBR-A11.bin
HBA2: SBR-A21.bin → LSI-P20-2118it.bin → LSI-P20-mptsas2.rom → SBR-A21.bin

Both port start to works!

**Comment by WhoAmI0501** [comment]
I had the same issue with my Fujitsu D2607-A21 controller. The SBR was modified as of a unknown reason while flashing the firmware, which lead to the result, that only half of the ports of my backplane worked.

Writing the SBR again after flashing the firmware solved the issue for me.

I may improve the documentation of this project a little bit and open a PR later... (@marcan)
